### DISCUSSION TITLE
Is it possible to "format" ECHO text? (location on the screen, font size and color?)

### DISCUSSION THREAD

**Discussion by GRIFFCOMMca** [ipxe/ipxe#1147] [Q&A]
Hi, we are creating a small booting script and have a few error trap tests (if failure) and would like to echo out "TFTP problem" in the middle of the screen in a large font, i cant find any documentation on how to do this, is it possible to format any ECHO text?

Couple that would be removing the text on the screen with out wiping the whole screen (although not a huge issue if a complete screen wipe is required).

We did search to see if this has asked before.....

**Comment by mcb30** [comment]
iPXE supports several ANSI escape sequences to control cursor positioning, text colour, etc.

There is no way to change font size, but you can control many other aspects of the text appearance.

You could also use an embedded background image to display your large fixed error message. You can use the `console` command to control where text subsequently appears on top of the image.

**Reply by GRIFFCOMMca** [reply]
Hi, are there any examples of text control code, we cant find anything on the net.  So far we are reloading a new background "picture" with the errors, it seems really inefficient though.

**Reply by mcb30** [reply]
You can find lots of examples in the Wikipedia page at https://en.wikipedia.org/wiki/ANSI_escape_code

To generate these in iPXE, you can use something like:
```
set esc:hex 1b            # ANSI escape character - "^["
set cls ${esc:string}[2J  # ANSI clear screen sequence - "^[[2J"
echo ${cls}
```
